traitsui/wx/list_editor.py

```diff
@@ -343,8 +345,19 @@ def add_item(self, offset):
         list, index = self.get_info()
         index += offset
         item_trait = self._trait_handler.item_trait
-        value = item_trait.default_value_for(self.object, self.name)
-        self.value = list[:index] + [value] + list[index:]
+        if self.factory.default_factory:
+            value = self.factory.default_factory(
+                *self.factory.default_args, **self.factory.default_kwargs
+            )
+        else:
+            value = item_trait.default_value_for(self.object, self.name)
+        try:
+            self.value = list[:index] + [value] + list[index:]
+        # if the default new item is invalid, we just don't add it to the list.
+        # traits will still give an error message, but we don't want to crash
+        except TraitError:
+            from traitsui.api import raise_to_debug
+            raise_to_debug()
```

Comment:
question : looks like we missed this in the earlier PR

Reply:
yeah wx got neglected before 😞